### ISSUE TITLE
defineProps in pure type syntax not working

### ISSUE THREAD

**Issue by boonyasukd** [johnsoncodehk/volar#26]
Hi, I recently tried out the new `<script setup>` syntax in Typescript and found out that things start to get a bit wonky when I tried to use `defineProps()` in pure type syntax. For example, from my system, Volar fails to recognize the following props definition:

```vue
<script lang="ts" setup>
const props = defineProps<{
  valid: Ref<boolean>;
  save: () => void;
  reset: () => void;
}>();
</script>
```

It's kinda hard to describe in words, here is how it looks on my project:
<img width="786" alt="Screen Shot 2020-11-27 at 01 54 07" src="https://user-images.githubusercontent.com/8109816/100384749-8105ec80-3053-11eb-8abf-9fae41ff4e80.png">

Volar underlined two `@click` definitions because it couldn't recognize the `save()` and `reset()` function in `defineProps()`. And somehow, after that fails, Volar also fails to recognize the `msg()` function as well.

**Comment by johnsoncodehk** [comment]
Thanks for your feedback!

I'm not sure this is a bug or not, I just parse `defineProps<{ ... }>()` to `defineComponent<{ ... }>()`, so the `void` type of `save()` / `reset()` is come from `defineComponent()`.

but you can do this:
```
defineProps({
  save: { type: Object as PropType<() => void>, required: true }
  reset: { type: Object as PropType<() => void>, required: true }
})
```

it will parse to `defineComponent({ props: { ... } })`, so you can have a predictable result.

the `msg()` problem like a type error, I need the error message to check.

**Comment by boonyasukd** [comment]
Hi, thanks for the quick response. I just tried the non pure-type way that you said, and all the errors now disappeared. :)

<img width="786" alt="Screen Shot 2020-11-27 at 03 28 17" src="https://user-images.githubusercontent.com/8109816/100389688-9df4ec80-3060-11eb-9aa9-bbffd0697a7e.png">

After taking a look at the spec again regarding [pure-type declaration](https://github.com/vuejs/rfcs/blob/script-setup-2/active-rfcs/0000-script-setup.md#type-only-propsemit-declarations), I just realize that "As of now, the type declaration argument must be a literal type to ensure correct static analysis." So I guess for function declaration (which isn't a literal type), I probably can't use pure-type declaration yet. :(

**Comment by boonyasukd** [comment]
At any rate, I just tried to refactored another component, and found out that even on a literal type, pure-type declaration seems to make Volar fails to recognize all declarations in the component. Switching to the object-based declaration (by commenting/uncommenting) and all errors just go away like the component I showed above.

<img width="786" alt="Screen Shot 2020-11-27 at 03 40 29" src="https://user-images.githubusercontent.com/8109816/100390402-d1387b00-3062-11eb-8d9a-ddcddb25afdc.png">

**Comment by johnsoncodehk** [comment]
It is seem a bug, `defineComponent<{ ... }>()` will broke the component context, will be fixed in next version :)

**Comment by johnsoncodehk** [comment]
fixed in 0.15.14.
